src/main/java/at/ac/tuwien/kr/alpha/common/atoms/FixedInterpretationLiteral.java

```diff
@@ -1,14 +1,42 @@
+/**
+ * Copyright (c) 2017-2018, the Alpha Team.
+ * All rights reserved.
+ *
+ * Additional changes made by Siemens.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1) Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2) Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
 package at.ac.tuwien.kr.alpha.common.atoms;
 
 import at.ac.tuwien.kr.alpha.grounder.Substitution;
 
 import java.util.List;
 
 /**
- * Represents an atom whose ground truth value(s) are independent of the current assignment.
- * Examples of such atoms are builtin atoms and external atoms.
- * Copyright (c) 2017, the Alpha Team.
+ * Represents a literal whose ground truth value(s) are independent of the current assignment.
+ * Examples of atoms underlying such literals are builtin atoms and external atoms.
+ * Copyright (c) 2017-2018, the Alpha Team.
  */
-public interface FixedInterpretationLiteral extends Literal {
+public interface FixedInterpretationLiteral {
```

Comment:
Is there a reason why `FixedInterpretationLiteral` no longer extends `Literal` (and becomes an abstract class by that)? I'm asking, because this now in principle allows that something is a `FixedInterpretationLiteral` but not a `Literal`.

Reply:
The only reason is that `Literal` became a class and an interface cannot extend a class. I´m not sure if one of the two design alternatives is better, but since you seem to have a preference towards `FixedInterpretationLiteral` extending the abstract class, I implemented this in e089a5dd3317cf8fc410c0376350d82da3538faf.